src/plugin/proto.rs

```diff
@@ -0,0 +1,250 @@
+use serde::{Deserialize, Serialize};
+
+pub type GitRevision = String;
+
+pub type Null = ();
+
+pub type ReleaseNotes = String;
+
+pub type MethodName = String;
+
+pub type Warning = String;
+
+pub type Error = String;
+
+#[derive(Clone, Debug)]
+pub struct Version {
+    pub rev: GitRevision,
+    pub semver: Option<semver::Version>,
+}
+
+pub mod request {
+    use super::*;
+    use crate::config::CfgMap;
+    use std::collections::HashMap;
+
+    pub struct PluginRequest<T> {
+        pub cfg_map: CfgMap,
+        pub env: HashMap<String, String>,
```

Comment:
Maybe enviroment should be isolated for different plugins?

Reply:
It could be, but why? This env is read-only for plugins.

Reply:
How to know that it is really read-only?

Reply:
You mean what if plugin writes to `std::env`? Yeah, indeed, that's a vulnerability.
I'll make an issue for that, we should store original env and restore it after calling a plugin.

Reply:
https://github.com/etclabscore/semantic-rs/issues/31